lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -2099,7 +2099,7 @@ class MongoClientInterface {
         let bucketCount = 0;
         const bucketInfos = [];
 
-        this.db.listCollections().toArray().then(collInfos =>
+        this.db.listCollections({ filter: { type: 'collection' } }).toArray().then(collInfos =>
```

Comment:
Is this function used in prod, not a dead code? It's used by arsenal from "countItems" and "scanItemCount", but we do not use these functions in s3utils anymore IIRC, we use https://github.com/scality/s3utils/blob/development/1.14/utils/S3UtilsMongoClient.js#L849 instead. I believe this code is here since the integration of Armory.
If we can confirm these are not used (beyond some tests in cloudserver), we should just drop them.

Reply:
It is used in Cloudserver's "metrics" pod

Reply:
(And the copy from s3utils should be removed to dedup' the code, relying on the single implementation here)